lib/SIL/SILConstants.cpp

```diff
@@ -882,7 +882,12 @@ void SymbolicValue::emitUnknownDiagnosticNotes(SILLocation fallbackLoc) {
   bool emittedFirstNote =
       emitNoteDiagnostic(badInst, getUnknownReason(), fallbackLoc, error);
 
+  auto sourceLoc = fallbackLoc.getSourceLoc();
   auto &module = badInst->getModule();
+  if (sourceLoc.isInvalid()) {
+    diagnose(module.getASTContext(), sourceLoc, diag::constexpr_not_evaluable);
```

Comment:
Doesn't this emit the new diagnostic at a known-to-be-invalid location (which will come out as unknown:0:0)?  Isn't there a better location we can use?

Reply:
There should be, but I have not looked into it. This patch just fixes a crasher. I filed [SR-8486](https://bugs.swift.org/browse/SR-8486).